### PR TITLE
Add inductive inference regression tests and fix reachability propagation

### LINKED ISSUE #196: Add unit tests for inductive inference

### PR DESCRIPTION
**Pull request by conorheins**
## Summary
- fix backward reachability propagation in `generate_I_matrix(...)` so predecessor states become reachable across induction depth
- add a dedicated `test/test_inductive_inference_jax.py` module covering reachability structure, exact inductive values, one-step ranking, disabled-inductive parity, and the current `qs_init`-anchored multistep semantics
- keep the existing policy-scoring rollout rule and agent-side `I` recomputation behavior unchanged

## Validation
- `uv run --group test python -m pytest test/test_inductive_inference_jax.py -q`
- `uv run --group test python -m pytest test/test_control_jax.py -k 'inductive or param_posterior' -q`\n- `uv run --group test python -m pytest test/test_agent_jax.py -k inductive -q`

## Demo impact
- the visible outputs of `examples/inductive_inference/inductive_inference_example.ipynb` are unchanged for the seeded example (`q_pi` and `neg_efe` stay the same)
- the visible rollouts in `examples/inductive_inference/inductive_inference_gridworld.ipynb` are unchanged for both notebook scenarios
- one internal `I` entry changes in the simple notebook example, but it does not affect the shown outputs for that seed and policy set

Closes #196
Refs #195

### CODE REVIEW

File: pymdp/control.py [hunk 1]
```diff
@@ -840,7 +840,10 @@ def update_posterior_policies_inductive(
 
 def generate_I_matrix(H: list[Array], B: list[Array], threshold: float, depth: int) -> list[Array]:
     """ 
-    Generates the `I` matrices used in inductive planning. These matrices stores the probability of reaching the goal state backwards from state j (columns) after i (rows) steps.
+    Generate inductive reachability matrices using backward state reachability.
+
+    These matrices store whether state `j` (columns) can still reach the
+    intended state set after `i` backward steps (rows).
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/control.py:848]: I changed the `generate_I_matrix(...)` docstring so it consistently describes I as a 0/1 reachability indicator in 591924a

File: pymdp/control.py [hunk 2]
```diff
@@ -868,16 +871,19 @@ def generate_I_matrix(H: list[Array], B: list[Array], threshold: float, depth: i
         For each factor, we need to compute the probability of reaching the goal state
         """
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/control.py]: removed the no-op triple-quoted string inside the loop in favor of a normal comment in 591924a

File: pymdp/agent.py [hunk 3]
```diff
@@ -277,6 +277,8 @@ def __init__(
 
         if self.inference_horizon is not None and self.inference_horizon < 1:
             raise ValueError("`inference_horizon` must be >= 1 when provided")
+        if self.inductive_depth < 1:
+            raise ValueError("`inductive_depth` must be >= 1")
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/agent.py]: Good catch, added the additional conditioning on `use_inductive` in 8ea7ae8
